contentgrid-spring-data-rest/src/main/java/com/contentgrid/spring/data/rest/validation/AllowedValues.java

```diff
@@ -0,0 +1,61 @@
+package com.contentgrid.spring.data.rest.validation;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.CONSTRUCTOR;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE_USE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import com.contentgrid.spring.data.rest.validation.AllowedValues.List;
+import jakarta.validation.Constraint;
+import jakarta.validation.Payload;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * The annotated element must have one of the allowed options as value.
+ */
+@Target({ METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER, TYPE_USE })
+@Retention(RUNTIME)
+@Repeatable(List.class)
```

Comment:
It doesn't really make sense to have this constraint as `@Repeatable`.

If you have a set of allowed values, adding an additional annotation will further restrict the list of allowed values; at which point you might as well have a single annotation containing the exact list of allowed values.

Additionally, for supporting multiple annotations, you would need to evaluate the resulting restriction to generate the proper HAL-FORMS options.

Reply:
Removed last commit